### ISSUE TITLE
Request headers are case insensitive but chai-nock is not

### ISSUE THREAD

**Issue by maciej-wakula** [chrisandrews7/chai-nock#11]
https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html

According to this document request headers are case insensitive.
When executing an example from documentation:
```it('requestedWithHeadersMatch', () => {
  const requestNock = nock('http://bbc.co.uk')
    .get('/')
    .reply(200);
 
  request({
    json: true,
    uri: 'http://bbc.co.uk',
    headers: {
      myHEADER: 'myHeaderValue',  /////////////////////////// Notice upper case here
      otherHeader: 'otherHeaderValue'
    }
  });
 
  return expect(requestNock).to.have.been.requestedWithHeadersMatch({
    myHeader: 'myHeaderValue'
  });
});```

it would fail as the verification is not case insensitive.

**Comment by adderek** [comment]
The deep-equal seems too much for simple equality check of headers. In fact Map() seems to fit better to the list of headers.
https://nodejs.org/api/http.html#http_response_getheaders is returning an Object though with keys in lowercase. And we cannot be sure what implementation users are using thus **I believe** that shallow equality should be checked only for headers and in an case insensitive fields manner - instead of `equals` from `deep-equal` - any more complex object should be either reported as error/failure or deep members ignored.
There should be also a way to check body+headers together - I have not seen it here.

Anyway - I liked the idea of this module -thanks for creating it.

**Comment by chrisandrews7** [comment]
Agreed @adderek. Feel free to raise a PR @adderek or @maciej-wakula-opuscapita.
